classpath-api-stubs/src/main/kotlin/net/msrandom/stubs/StubGenerator.kt

```diff
@@ -117,7 +120,7 @@ object StubGenerator {
         node.version = min(nodeA.version, nodeB.version)
         node.access = nodeA.access
         node.name = nodeA.name
-        node.signature = nodeA.signature
+        node.signature = nodeB.signature?.let { nodeA.signature?.commonPrefixWith(it) } ?: nodeA.signature
```

Comment:
The fallback `nodeA.signature` is not needed, as it should be null if one of the nodes is null